core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -53,172 +92,185 @@ fun findPath(
-        makeSignalingSystemConstraints(
-            fullInfra.blockInfra,
-            fullInfra.signalingSimulator,
-            listOf(rollingStock)
-        )
+    fun findPath(): STDCMResult? {
+        assert(steps.size >= 2) { "Not enough steps have been set to find a path" }
+
+        // Initialize the A* heuristic
+        val locations = steps.stream().map(STDCMStep::locations).toList()
+        val remainingDistanceEstimators = makeHeuristics(fullInfra, locations)
+        estimateRemainingDistance = makeAStarHeuristic(remainingDistanceEstimators, rollingStock)
 
-    // Initialize the A* heuristic
-    val locations = steps.stream().map(STDCMStep::locations).toList()
-    val remainingDistanceEstimators = makeHeuristics(fullInfra, locations)
-    val endBlocks = steps.last().locations.map { it.edge }
-    val path =
-        Pathfinding(graph)
-            .setEdgeToLength { edge -> edge.infraExplorer.getCurrentBlockLength().cast() }
-            .setRemainingDistanceEstimator(
-                makeAStarHeuristic(remainingDistanceEstimators, rollingStock)
+        val endBlocks = steps.last().locations.map { it.edge }
+        starts =
+            convertLocations(
+                graph,
+                steps[0].locations,
+                startTime,
+                maxDepartureDelay,
+                rollingStock,
+                endBlocks
             )
-            .setEdgeToLength { edge -> edge.length.cast() }
-            .addBlockedRangeOnEdges { edge: STDCMEdge ->
-                convertRanges(loadingGaugeConstraints.apply(edge.block))
+        val path = findPathImpl() ?: return null
+
+        return STDCMPostProcessing(graph)
+            .makeResult(
+                fullInfra.rawInfra,
+                path,
+                startTime,
+                graph.standardAllowance,
+                rollingStock,
+                timeStep,
+                comfort,
+                maxRunTime,
+                blockAvailability
+            )
+    }
+
+    private fun findPathImpl(): Result? {
+        val queue = PriorityQueue<STDCMEdge>()
+        for (location in starts) {
+            val len = location.length
+            val totalCostUntilEdge = computeTotalCostUntilEdge(location, len)
+            val distanceLeftEstimation =
+                // TODO: inconsistent, same as Pathfinding
+                estimateRemainingDistance!![0].apply(location, Offset(0.meters))
+            location.weight = distanceLeftEstimation + totalCostUntilEdge
+            queue.add(location)
+        }
+        val start = Instant.now()
+        while (true) {
+            if (Duration.between(start, Instant.now()).toSeconds() >= pathfindingTimeout)
+                throw OSRDError(ErrorType.PathfindingTimeoutError)
+            val edge = queue.poll() ?: return null
+            // TODO: we mostly reason in terms of endNode, we should probably change the queue.
+            val endNode = edge.getEdgeEnd(graph)
```

Comment:
It's functionally the same, but semantically we use the graph interface

```suggestion
            val endNode = graph.getEdgeEnd(edge)
```

Reply:
done